cmd/sink/filters/filters.go

```diff
@@ -128,34 +127,83 @@ func (f *Filters) createFilters(
 	if err != nil {
 		return archiveMap, deleteMap, archiveOnDeleteMap, err
 	}
+
 	var errList []error
 	for _, resource := range resources {
 		gvk := schema.FromAPIVersionAndKind(resource.Selector.APIVersion, resource.Selector.Kind)
 		namespaceGvk := NamespaceGVKFromNamespaceAndGvk(namespace, gvk)
 		if resource.ArchiveWhen != "" {
-			archiveExpr, err := ocel.CreateCelExprOr(resource.ArchiveWhen, globalArchive[gvk.String()])
+			archiveExpr, err := ocel.CreateCelExprOr(resource.ArchiveWhen, globalArchive[gvk])
 			if err != nil {
 				errList = append(errList, err)
 			} else {
 				archiveMap[namespaceGvk] = *archiveExpr
 			}
 		}
 		if resource.DeleteWhen != "" {
-			deleteExpr, err := ocel.CreateCelExprOr(resource.DeleteWhen, globalDelete[gvk.String()])
+			deleteExpr, err := ocel.CreateCelExprOr(resource.DeleteWhen, globalDelete[gvk])
 			if err != nil {
 				errList = append(errList, err)
 			} else {
 				deleteMap[namespaceGvk] = *deleteExpr
 			}
 		}
 		if resource.ArchiveOnDelete != "" {
-			archiveOnDeleteExpr, err := ocel.CreateCelExprOr(resource.ArchiveOnDelete, globalArchiveOnDelete[gvk.String()])
+			archiveOnDeleteExpr, err := ocel.CreateCelExprOr(resource.ArchiveOnDelete, globalArchiveOnDelete[gvk])
 			if err != nil {
 				errList = append(errList, err)
 			} else {
 				archiveOnDeleteMap[namespaceGvk] = *archiveOnDeleteExpr
 			}
 		}
+
+		// Now we cycle the global filters `globalArchive`, `globalDelete` and `globalArchiveOnDelete`
```

Comment:
The three loops below are identical except for the global and local maps.  Why not split them out into a private method that can be called for each of the global/local map pairs?

Reply:
Done.